### ISSUE TITLE
We ship with JBossRules Firing

### ISSUE THREAD

**Issue by wezell** [dotCMS/core#7681]
Seen in a thread dump running from distro:

```
"url:GET//demo.dotcms.com/ | lang:1 | ip:127.0.0.1 | Admin:false | start:05-01-2015 09:15:34 EDT" daemon prio=10 tid=0x0000000000a8d800 nid=0x12eb runnable [0x00007f28c8fcd000]
   java.lang.Thread.State: RUNNABLE
	at java.util.WeakHashMap.put(WeakHashMap.java:521)
	at com.dotcms.repackage.org.drools.common.AbstractRuleBase.addWorkingMemory(Unknown Source)
	at com.dotcms.repackage.org.drools.reteoo.ReteooRuleBase.newWorkingMemory(Unknown Source)
	at com.dotcms.repackage.org.drools.common.AbstractRuleBase.newWorkingMemory(Unknown Source)
	at com.dotmarketing.util.JBossRulesUtils.checkObjectRulesFromXML(JBossRulesUtils.java:161)
	at com.dotmarketing.filters.CMSFilter.doFilter(CMSFilter.java:213)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:239)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:206)
	at com.dotmarketing.filters.AutoLoginFilter.doFilter(AutoLoginFilter.java:61)
```

Looking at the code, it is calling
https://github.com/dotCMS/core/blob/master-3.2/src/com/dotmarketing/util/JBossRulesUtils.java#L160
and XMLRuleBase is set and !=null

**Comment by wezell** [comment]
Actually, I am doing benchmarks and all the threads are in this method.  Not sure what it is doing but it can't be good.

**Comment by oarrietadotcms** [comment]
@wezell  what is the purpose of JBossRulesUtils?

https://github.com/dotCMS/core/blob/master-3.2/src/com/dotmarketing/filters/CMSFilter.java#L213

**Comment by dotcmsCodeReview** [comment]
Long story, but needless to say, the new rules engine will replace it.  I
would just remove it from the CMSFilter - if people want to use the old
JBoss rules functionality, they can uncomment the JBossRulesFilter in their
web.xml

On Fri, May 8, 2015 at 10:57 AM, Oscar Arrieta <notifications@github.com>
wrote:

> @wezell <https://github.com/wezell> what is the purpose of
> JBossRulesUtils?
>
>
> https://github.com/dotCMS/core/blob/master-3.2/src/com/dotmarketing/filters/CMSFilter.java#L213
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/dotCMS/core/issues/7681#issuecomment-100261992>.
>



-- 






3059 Grand Avenue
Suite 410-B
Miami FL 33133
Main: 305-900-2001 | Direct: 978.294.9429

will@dotcms.com | dotcms.com <http://www.dotcms.com/>

<https://twitter.com/dotcms>  <http://www.linkedin.com/company/dotcms>
<https://plus.google.com/+Dotcms>  <https://www.facebook.com/dotCMS>

**Comment by oarrietadotcms** [comment]
got it.

**Comment by oarrietadotcms** [comment]
PR: https://github.com/dotCMS/core/pull/7715

**Comment by bryanboza** [comment]
Fixed, tested on release 3.2.2 // Postgres // FF
